src/components/sync-connected-sites.tsx

```diff
@@ -54,13 +63,41 @@ export const convertTreeToOptionsToSync = ( tree: TreeNode[] ): SyncOption[] =>
 		const wpContent = filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
 
 		wpContent.forEach( ( item ) => {
-			if ( item.checked && isSyncOption( item.id ) ) {
-				optionsToSync.push( item.id );
+			if ( isSyncOption( item.id ) ) {
+				const hasCheckedChildren = item.children?.some( ( child ) => child.checked );
+				const shouldInclude = item.checked || hasCheckedChildren;
+
+				if ( shouldInclude ) {
+					optionsToSync.push( item.id );
+
+					if ( item.id === SYNC_OPTIONS.plugins && item.children ) {
+						const selectedPlugins = item.children
+							.filter( ( plugin ) => plugin.checked )
+							.map( ( plugin ) => plugin.id );
+						if ( selectedPlugins.length > 0 && selectedPlugins.length < item.children.length ) {
+							specificSelections.plugins = selectedPlugins;
+						}
+					}
+
+					if ( item.id === SYNC_OPTIONS.themes && item.children ) {
+						const selectedThemes = item.children
+							.filter( ( theme ) => theme.checked )
+							.map( ( theme ) => theme.id );
+						if ( selectedThemes.length > 0 && selectedThemes.length < item.children.length ) {
+							specificSelections.themes = selectedThemes;
+						}
+					}
+				}
 			}
 		} );
 	}
 
-	return optionsToSync;
+	const hasSpecificSelections = specificSelections.plugins || specificSelections.themes;
+
+	return {
+		optionsToSync,
+		specificSelections: hasSpecificSelections ? specificSelections : undefined,
+	};
```

Comment:
WDYT about removign extra 2 conditions for `hasSpecificSelections` with defining undefined by default `let specificSelections: { plugins?: string[]; themes?: string[] } | undefined;` and then we can set children with parent object together, for example `specificSelections = { ...specificSelections, plugins: checkedChildren };`


```suggestion
	return {
		optionsToSync,
		specificSelections,
	};
```